### PR TITLE
fix(geolocator): Stop using location when unsubscribing  from events.

### LINKED ISSUE #11915: [Droid] Geolocator doesn't stop location using
### Current behavior

Started location reading and stop by unsubscribing from events of geolocator Uno continues using device's location. The location icon presents in the bar

photo_2023-04-06_15-46-58

### Expected behavior

Stop location use

### How to reproduce it (as minimally and precisely as possible)

Repro
[LocationReadingIssues.zip](https://github.com/unoplatform/uno/files/11168862/LocationReadingIssues.zip)

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

Uno 4.8.15

### Affected platforms

Android

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Robert-Louis**
GitHub Issue (If applicable): closes #11915 

## PR Type

What kind of change does this PR introduce?

- Bugfix
## What is the current behavior?

- when unsubscribing from events the continue to use the location.

## What is the new behavior?

-  Stop using location when unsubscribing  from events.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs [hunk 1]
```diff
@@ -397,6 +401,7 @@ private void RemoveUpdates()
 			&& _locationListener.Handle != IntPtr.Zero)
 		{
 			_locationManager.RemoveUpdates(_locationListener);
+			BroadcastStatusChanged(PositionStatus.Disabled);
```
Comment (Youssef1313) [inline src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:404]: I'm not sure about this change. Why it is needed?

File: src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs [hunk 2]
```diff
@@ -310,7 +310,11 @@ private async void InitializeIfPermissionIsGranted()
 		RestartUpdates();
 	}
 
-	partial void StopPositionChanged() => _positionChangedSubscriptions.TryRemove(this, out var _);
+	partial void StopPositionChanged()
+	{
+		_positionChangedSubscriptions.TryRemove(this, out var _);
```
Comment (Youssef1313) [inline src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:315]:
```suggestion
		_positionChangedSubscriptions.TryRemove(this, out _);
```

File: src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs (hunk 1 above)
Comment (Robert-Louis) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:404]: We don't update the localisation anymore a this point. It's seem weird when Status Changed is attached and when we Disabled the localisation that it's not updated. 

If unnacessary I can removed this changed

Comment (Xiaoy312) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:404]: RemoveUpdates() is called in many places, not sure if that is the right choice?
What you could do is to BroadcastStatusChanged(PositionStatus.Disabled) in StopPositionChanged()

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs [hunk 3]
```diff
@@ -310,7 +310,12 @@ private async void InitializeIfPermissionIsGranted()
 		RestartUpdates();
 	}
 
-	partial void StopPositionChanged() => _positionChangedSubscriptions.TryRemove(this, out var _);
+	partial void StopPositionChanged()
+	{
+		_positionChangedSubscriptions.TryRemove(this, out _);
+		RemoveUpdates();
+		BroadcastStatusChanged(PositionStatus.Disabled);
```
Comment (Youssef1313) [inline src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:317]: @jeromelaban I think https://github.com/unoplatform/uno/pull/12091#pullrequestreview-1404924397 needs to be addressed first. Making this comment to temporary block auto merge.

Comment (Youssef1313) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:317]: Oh actually I can no longer make any sense of my comment 😄 
Resolving this for now. I might have read something wrong.

**Pull request merged**